src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -6624,11 +6626,13 @@ test_bypass_mongocryptd_shared_library (void *unused)
       auto_encryption_opts, "keyvault", "datakeys");
 
    // wait for port and ip to be set on the other thread
-   while (!args->port) {
-      int ret = mongoc_cond_timedwait (&args->cond, &args->mutex, 5000);
+   bson_mutex_lock (&args->mutex);
+   while (!args->port || 0 == strlen (args->ip)) {
```

Comment:
Suggest moving `0 == strlen (args->ip)` out of loop and into a subsequent `BSON_ASSERT`:
```c
// wait for port and ip to be set on the other thread
bson_mutex_lock (&args->mutex);
while (!args->port) {
   int cond_ret = mongoc_cond_timedwait (&args->cond, &args->mutex, 5000);
   /* ret non-zero indicates an error (a timeout) */
   BSON_ASSERT (!cond_ret);
}
bson_mutex_unlock (&args->mutex);
BSON_ASSERT (strlen (args->ip) > 0); // <--
```